### PR TITLE
feat: updating implementation of 104 zalando guideline (#610)

### LINKED ISSUE #610: Adopt the Changes of Zalando Rule [104]
**Acceptance Criteria**
- Title is updated
- OAuth2 service-to-service communication flow must be `application`

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #610.

### CODE REVIEW

**Comment by netme** [comment]
@maxim-tschumak could you please take a look on integration test?

```
--- FAIL: TestIntegrationWithZallyApiDefinition (0.67s)
    --- FAIL: TestIntegrationWithZallyApiDefinition/integrationWithZallyApiDefinition (0.67s)
	Error Trace:	integration_test.go:140
	Error:      	"SHOULD
	            	======
	            	
	            	
SHOULD
 
Set Flow to 'application' When Using OAuth 2.0
	            		OAuth2 security definitions should use application flow
	            		https://zalando.github.io/restful-api-guidelines/#104
	            	
	            	
SHOULD
 
Use Specific HTTP Status Codes
	            		Operatons should use specific HTTP status codes
	            		https://zalando.github.io/restful-api-guidelines/#150
	            			/api-violations POST default
	            			/review-statistics GET default
	            			/supported-rules GET default
	            	
	            	Summary:
	            	========
	            	
	            	MUST violations: 0
	            	SHOULD violations: 2
	            	MAY violations: 0
	            	HINT violations: 0
	            	" does not contain "SHOULD violations: 1"
```

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/main/java/de/zalando/zally/rule/zalando/SecureWithOAuth2Rule.kt [hunk 1]
```diff
@@ -15,34 +15,34 @@ import org.springframework.stereotype.Component
...
-                .filter { (it as OAuth2Definition).flow != "password" }
+            .securityDefinitions
+            .orEmpty()
+            .values
+            .filter { it.type?.toLowerCase() == "oauth2" }
+            .filter { (it as OAuth2Definition).flow != "application" }
 
         return if (definitionsWithoutPasswordFlow.any())
-            Violation(this, "Set Flow to Password When Using OAuth 2.0", "OAuth2 security definitions should use password flow", ViolationType.SHOULD, emptyList())
+            Violation(this, "Set Flow to 'application' When Using OAuth 2.0", "OAuth2 security definitions should use application flow", ViolationType.SHOULD, emptyList())
```
Comment (maschleg) [inline server/src/main/java/de/zalando/zally/rule/zalando/SecureWithOAuth2Rule.kt]: typo: ``when using`` lowercase

Comment (maschleg) [reply server/src/main/java/de/zalando/zally/rule/zalando/SecureWithOAuth2Rule.kt]: and I would stick to the same wording ``OAuth 2.0``-> ``OAuth2``

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by maschleg** [comment]
👍

**Comment by netme** [comment]
👍

**Pull request merged**